File: src/main/java/com/example/jpa_lab/schemas/Product.java
package com.example.jpa_lab.schemas;

import jakarta.persistence.*;

@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int productId;
    private String productName;
    private int unitsInStock;


    @ManyToOne(fetch = FetchType.LAZY)
    private Supplier supplier;

    @ManyToOne(fetch = FetchType.LAZY)
    private Category category;
    public Product(){

    }

    public Product(String productName, int unitsInStock, Supplier supplier, Category category) {
        this.productName = productName;
        this.unitsInStock = unitsInStock;
        this.supplier = supplier;
        this.category = category;
    }

}

File: entity_framework/entityFrameworkLab/Product.cs
public class Product
{
    public int ProductID { get; set;}
    public String? ProductName { get; set;}
    public int UnitsOnStock { get; set;}
    public int SupplierID { get; set;}

    public Supplier Supplier { get; set; } = null!;
    public ICollection<Invoice> Invoices { get; set; } = new List<Invoice>();
}